app/hooks/use-diagnostic-permission-prompt.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+import { useHistory } from 'react-router';
+
+import routes from '../constants/routes.json';
+import { useHasUserGivenDiagnosticPermissions } from '@store/settings';
+
+export function usePromptUserToSetDiagnosticPermissions() {
+  const history = useHistory();
+  const hasGivenPermission = useHasUserGivenDiagnosticPermissions();
+
+  useEffect(() => {
+    if (hasGivenPermission === undefined) history.push(routes.REQUEST_DIAGNOSTICS);
+    // Only want to run on mount, not every time this value changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [history]);
```

Comment:
I don't understand this comment, then why wouldn't you just leave the dep array empty?

Reply:
because then the linter complains that I'm not declaring all my dependencies

Reply:
Got it, thx!